lib/reporting_client/platforms/datadog.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'datadog/statsd'
+
+require_relative './base'
+
+module ReportingClient
+  module Platforms
+    class Datadog < Base
+      attr_reader :name
+
+      def initialize(name:, payload:)
+        @name = name
+
+        super(payload:)
+      end
+
+      def call
+        create_event(name, payload)
+      end
+
+      private
+
+      def create_event(name, payload = {})
+        return unless payload.delete(:instrument_datadog)
+
+        payload[:title] = name
+        event_name = payload.delete(:event_name) || name
+        payload[:event_name] = event_name
+
+        statsd = ::Datadog::Statsd.new(config.datadog_host, 8125,
+                                       namespace: config.datadog_service,
+                                       tags: ["env:#{config.datadog_env}", "service:#{config.datadog_service}"])
+
+        statsd.event(event_name, event_name, tags: build_tags(payload))
```

Comment:
```suggestion
        statsd.event(event_name, payload.to_json, tags: build_tags(payload), source_type_name: #{config.datadog_service})
```
After tinkering around a bit trying to make charts and such, I think it'd be more user friendly to have the JSONified payload be sent in as the `message` vs just repeating the event_name.  Especially given that we're moving away from `UserEvent` in DD.

Also, I'd like to try sending in `source_type_name` so we can get rid of `my_apps`